lib/detectors/UnderConstraints/UnderConstraints.h

```diff
@@ -0,0 +1,90 @@
+//
+// Created by Daniel Dominguez on 15/11/2022.
+//
+
+#ifndef VANGUARD_UNDERCONSTRAINTS_H
+#define VANGUARD_UNDERCONSTRAINTS_H
+
+#include "../UniverseDetector.h"
+#include "analysis/interproc-analysis/InterProceduralAnalyses.h"
+#include "analysis/taint-analysis/TaintAnalysis.h"
+#include "domain/libCircuit/CircuitModel.h"
+
+namespace vanguard {
+
+    template<typename Domain>
+    class UnderConstraints : public UniverseDetector<Domain> {
+    public:
+        std::vector<Requirement *> registerAnalyses() override {
+            return {};
+        }
+
+        void startDetection() override {
+            analysis = new TaintAnalysis();
+        }
+
+        void detect(typename Domain::Universe &universe) override {
+            for (auto unit : universe.units())
+                for (auto component : unit->components())
```

Comment:
nit: explicitly indicating pointer on auto helps with readability
```suggestion
            for (auto *unit : universe.units())
                for (auto *component : unit->components())
```

Reply:
I didn't knew you could do that. I will keep it in mind from now on